scripts/duplicate-dependencies.js

```diff
@@ -0,0 +1,85 @@
+// @ts-check
```

Comment:
theia cli tool has a command to check it, we should use it instead: https://github.com/eclipse-theia/theia/blob/d5a60bb069bdb126dcdf0166bc69cca1014eb291/dev-packages/cli/src/theia.ts#L125-L144

It is important, since downstream electron projects should be able to run the same check.

Reply:
The thing is, `electron-builder` seems to create its own dependency layout when bundling, which would be different from what `yarn` does.

Example: `yarn` was correctly hoisting `fs-extra`, but when bundled it was not. This made the app fail at runtime. The solution was to have whatever extension depending on `fs-extra` to explicitly declare so, then tooling was able to correctly install things.

My conclusion is that we should not rely on hoisting.

Reply:
It is not about electron-builder only. Someone develops a product and wants to check whether there is not duplicate dependencies then he should run exactly the same script which we run to check duplicate dependencies in this repo. `check:hoisted` is the script

cc @kittaakos

Reply:
If we want singleton packages, then we would have to deal with peer dependencies

https://classic.yarnpkg.com/blog/2018/04/18/dependencies-done-right/

I might open an issue later to discuss that more in detail, I really think that our current system is too fragile.

Reply:
I don't really care about peer dependencies. The point is that we don't need 2 scripts doing the same things. Please use `check:hoisted`.